lib/extensions/postgres_cdc_rls/replication_poller.ex

```diff
@@ -99,76 +94,27 @@ defmodule Extensions.PostgresCdcRls.ReplicationPoller do
     cancel_timer(poll_ref)
     cancel_timer(retry_ref)
 
-    try do
-      {time, response} =
-        :timer.tc(Replications, :list_changes, [
-          conn,
-          slot_name,
-          publication,
-          max_changes,
-          max_record_bytes
-        ])
-
-      Realtime.Telemetry.execute(
-        [:realtime, :replication, :poller, :query, :stop],
-        %{duration: time},
-        %{tenant: tenant}
+    broadcast_count =
```

Comment:
I feel like this whole thing including `list_changes_with_telemetry` could be a with statement with each step making it very obvious what's going on.

Reply:
Like break out what's going in `handle_list_changes_result`. Responding with the count of changes handled seems odd.

Reply:
will do 👍 that is the main priority for the code cleanup: be more clear on the flow of actions happening and that is a great suggestion